mw/prometheus/prometheus.go

```diff
@@ -0,0 +1,102 @@
+package prometheus
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/gojekfarm/ziggurat"
+	"github.com/gojekfarm/ziggurat/server"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const (
+	namespace        = "ziggurat_go"
+	handlerSubsystem = "handler"
+)
+
+const (
+	// RouteLabel - Key for route label
+	RouteLabel = "route"
+)
+
+// HandlerEventsCounter - Prometheus counter for handled events
+var HandlerEventsCounter = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Namespace: namespace,
+		Subsystem: handlerSubsystem,
+		Name:      "events_total",
+		Help:      "Events passed on to the handler, partitioned by route",
+	},
+	[]string{RouteLabel},
+)
+
+// HandlerFailuresCounter - Prometheus counter for handler failures
+var HandlerFailuresCounter = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Namespace: namespace,
+		Subsystem: handlerSubsystem,
+		Name:      "failures_total",
+		Help:      "Event handler failures, partitioned by route",
+	},
+	[]string{RouteLabel},
+)
+
+// HandlerDurationHistogram - Prometheus histogram for handler duration
+var HandlerDurationHistogram = prometheus.NewHistogramVec(
+	prometheus.HistogramOpts{
+		Namespace: namespace,
+		Subsystem: handlerSubsystem,
+		Name:      "duration_ms",
```

Comment:
It is best to standardize and use base units as seconds instead of ms. Let's change this to use seconds.